### ISSUE TITLE
RateLimitTransformer no more compatible with SDK dev.23

### ISSUE THREAD

**Issue by wem** [dart-lang/angular_components#220]
After update the Dart SDK to version dev-23.0 is looks like that this commit https://github.com/dart-lang/sdk/commit/0d5ce913c92b59da7ed9a15a6ca1250f7f039f35 has a breaking change for the StreamTransformer which result in:

```
[SEVERE] Instance of 'DevCompilerBuilder' on angular_components|lib/src/utils/async/rate_limit.module: 
Error compiling dartdevc module:angular_components|lib/src/utils/async/rate_limit.ddc.js

[error] Invalid override. The type of 'RateLimitTransformer.retype' ('() → StreamTransformer<S, T>') isn't a subtype of 'StreamTransformer<S, T>.retype' ('<RS,RT>() → StreamTransformer<RS, RT>'). (package:angular_components/src/utils/async/rate_limit.dart, line 59, col 3)
[error] Invalid override. The type of 'RateLimitTransformer.cast' ('() → StreamTransformer<dynamic, T>') isn't a subtype of 'StreamTransformer<S, T>.cast' ('<RS,RT>() → StreamTransformer<RS, RT>'). (package:angular_components/src/utils/async/rate_limit.dart, line 65, col 3)
[error] The name 'R' isn't a type so it can't be used as a type argument. (package:angular_components/src/utils/async/rate_limit.dart, line 65, col 21)

Please fix all errors before compiling (warnings are okay).
}
[WARNING] Instance of 'WebEntrypointBuilder' on ibex|test/client_only/components/table/xs_table_component_test.dart: Unable to read angular_components|lib/src/utils/async/rate_limit.ddc.js, check your console for compilation errors.
```

**Comment by nshahan** [comment]
This fix is landing internally and will be part of the next sync.

To unblock you for now you can edit your local version of https://github.com/dart-lang/angular_components/blob/master/lib/src/utils/async/rate_limit.dart#L46. 

Change:
```dart
class RateLimitTransformer<S, T> implements StreamTransformer<S, T> {
```
to:
```dart
class RateLimitTransformer<S, T> extends StreamTransformerBase<S, T> {
```

**Comment by nshahan** [comment]
The latest update includes this fix #222.
